### PR TITLE
PWN-4712 - Add API 33 integration for application

### PR DESCRIPTION
**Pull request by Davran331122**
## Jira Ticket

https://p2pvalidator.atlassian.net/browse/PWN-4712

## Description of Work

[ what work was done in this ticket ]

## Screenshots (Optional)

Screenshot #1 | Screenshot #2 | Screenshot #3
:------------:|:-------------:|:------------:
[Paste Pic 1] | [Paste Pic 2] | [Paste Pic 3]

### CODE REVIEW

File: app/src/main/java/org/p2p/wallet/common/mvp/BaseMvpFragment.kt [hunk 1]
```diff
@@ -24,9 +24,14 @@ abstract class BaseMvpFragment<V : MvpView, P : MvpPresenter<V>>(
         val view = super.onCreateView(inflater, container, savedInstanceState)?.apply {
             setOnClickListener { if (keyboardIsVisible) hideKeyboard() }
         }
+        onRequestPermission()
         return view
     }
 
+    protected open fun onRequestPermission() {
```
Comment (kabramovp2p) [inline app/src/main/java/org/p2p/wallet/common/mvp/BaseMvpFragment.kt:31]: = Unit 
 pls

File: app/src/main/java/org/p2p/wallet/android/PermissionExt.kt [hunk 2]
```diff
@@ -0,0 +1,86 @@
...
+
+import android.Manifest
+import android.os.Build
+import androidx.activity.result.contract.ActivityResultContracts
+import androidx.core.app.NotificationManagerCompat
+import androidx.fragment.app.Fragment
+import org.p2p.wallet.common.permissions.PermissionState
+import org.p2p.wallet.common.permissions.PermissionsUtil
+
+// TODO Replace all permission interaction in BaseQrFragment using this functions,
```
Comment (gslevinkov) [inline app/src/main/java/org/p2p/wallet/android/PermissionExt.kt:11]: todo needs a task

**Review by gslevinkov — APPROVED** [review]

File: app/src/main/AndroidManifest.xml [hunk 3]
```diff
@@ -12,7 +12,12 @@
     <uses-permission
         android:name="android.permission.WRITE_EXTERNAL_STORAGE"
         tools:ignore="ScopedStorage" />
-    <uses-permission android:name="android.permission.READ_EXTERNAL_STORAGE" />
+    <uses-permission
+        android:name="android.permission.READ_EXTERNAL_STORAGE"
+        android:maxSdkVersion="32" />
```
Comment (1sakov) [inline app/src/main/AndroidManifest.xml:17]: how to request external storage permission for 33?

Comment (Davran331122) [reply app/src/main/AndroidManifest.xml:17]: For 33 API you cannot request permission for external storage. You can only use scoped storage of your application,
Or if you want to pick image, you can use new MediaPicker with selecting media type

Comment (Davran331122) [reply app/src/main/AndroidManifest.xml:17]: If you need to read audio, declare also the android.permission.READ_MEDIA_AUDIO. For video, use the android.permission.READ_MEDIA_VIDEO permission. If the user has already given his permission to READ_EXTERNAL_STORAGE, the newly declared media permissions will automatically be granted when he updates his app to the new target level.
@1sakov

File: app/src/main/AndroidManifest.xml [hunk 4]
```diff
@@ -12,7 +12,12 @@
...
+
+    <uses-permission android:name="android.permission.POST_NOTIFICATIONS" />
+    <uses-permission android:name="android.permission.READ_MEDIA_IMAGES"/>
```
Comment (Davran331122) [inline app/src/main/AndroidManifest.xml:20]: Forgot to remove this

**Review by kabramovp2p — APPROVED** [review]

File: app/src/main/java/org/p2p/wallet/android/PermissionExt.kt [hunk 5]
```diff
@@ -0,0 +1,85 @@
+package org.p2p.wallet.android
```
Comment (1sakov) [inline app/src/main/java/org/p2p/wallet/android/PermissionExt.kt:1]: Can you name it `PermissionExtensions` please?

**Review by 1sakov — APPROVED** [review]
lgtm, gj

**Pull request merged**